luarules/gadgets/unit_death_animations.lua

```diff
@@ -14,38 +14,40 @@ if not gadgetHandler:IsSyncedCode() then
 	return
 end
 
-local hasDeathAnim = {
-  [UnitDefNames.corkarg.id] = true,
-  [UnitDefNames.corthud.id] = true,
-  [UnitDefNames.corstorm.id] = true,
-  [UnitDefNames.corsumo.id] = true,
-  [UnitDefNames.armraz.id] = true,
-  [UnitDefNames.armpw.id] = true,
-  [UnitDefNames.armck.id] = true,
-  [UnitDefNames.armrectr.id] = true,
-  [UnitDefNames.armrock.id] = true,
-  [UnitDefNames.armfast.id] = true,
-  [UnitDefNames.armzeus.id] = true,
-  [UnitDefNames.armfido.id] = true,
-  [UnitDefNames.armham.id] = true,
-  [UnitDefNames.corak.id] = true,
-  [UnitDefNames.corck.id] = true,
+local units = {
+	corkarg = true,
+	corthud = true,
+	corstorm = true,
+	corsumo = true,
+	armraz = true,
+	armpw = true,
+	armck = true,
+	armrectr = true,
+	armrock = true,
+	armfast = true,
+	armzeus = true,
+	armfido = true,
+	armham = true,
+	corak = true,
+	corck = true,
 }
-
-local dyingUnits = {}
-
-for udid, ud in pairs(UnitDefs) do --almost all raptors have dying anims
+for name,v in pairs(units) do
+	units[name..'_scav'] = true
+end
+local hasDeathAnim = {}
+for udid, ud in pairs(UnitDefs) do
+	if units[ud.name] then
+		hasDeathAnim[udid] = true
+	end
+	-- almost all raptors have dying anims
 	if string.find(ud.name, "raptor") or (ud.customParams.subfolder and ud.customParams.subfolder == "other/raptors") then
 		hasDeathAnim[udid] = true
 	end
 end
+units = nil
```

Comment:
This does nothing, the garbage collector will free the table as soon as the file is processed and it goes out of scope anyway.